docker/bin/zookeeperStart.sh

```diff
@@ -99,9 +98,35 @@ if [[ "$WRITE_CONFIGURATION" == true ]]; then
 
     echo Registering node and writing local configuration to disk.
     java -Dlog4j.configuration=file:"$LOG4J_CONF" -jar /root/zu.jar add $ZKURL $MYID  $ZKCONFIG $DYNCONFIG
-
   fi
 fi
 
+ZOOCFGDIR=/data/conf
+export ZOOCFGDIR
+
+if [ ! -d "$ZOOCFGDIR" ]
+then
+echo "Creating writable conf directory."
+mkdir $ZOOCFGDIR
+cp -f /conf/zoo.cfg $ZOOCFGDIR
+cp -f /conf/log4j.properties $ZOOCFGDIR
+cp -f /conf/log4j-quiet.properties $ZOOCFGDIR
+cp -f /conf/env.sh $ZOOCFGDIR
+fi
+
+if [[ "$WRITE_CONFIGURATION" == false &&  "$REGISTER_NODE" == false ]]; then
+  # We get here only on server restart...
+  echo Printing bootstrap dynamic config file
+  cat $DYNCONFIG
+  echo Printing Static Config
+  STATIC_CONFIG=`cat $ZOOCFGDIR/zoo.cfg`
+  echo Writing configuration gleaned from zookeeper ensemble
+  echo "$CONFIG" | grep -v "^version="> $DYNCONFIG
```

Comment:
One clarification question: `$CONFIG` is an input to the script, what's it expected to be, the last dynamic config file?

Reply:
The value in $CONFIG is generated by querying zk-service for the current dynamic configuration.
So it will always reflect the current dynamic configuration.